src/Functions/seriesOutliersTukey.cpp

```diff
@@ -0,0 +1,173 @@
+
+        return std::make_shared<DataTypeArray>(std::make_shared<DataTypeFloat64>());
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t) const override
+    {
+        ColumnPtr array_ptr = arguments[0].column;
+        const ColumnArray * array = checkAndGetColumn<ColumnArray>(array_ptr.get());
+
+        const IColumn & src_data = array->getData();
+        const ColumnArray::Offsets & src_offsets = array->getOffsets();
+
+        ColumnPtr res;
+
+        if (executeNumber<UInt8>(src_data, src_offsets, res) || executeNumber<UInt16>(src_data, src_offsets, res)
+            || executeNumber<UInt32>(src_data, src_offsets, res) || executeNumber<UInt64>(src_data, src_offsets, res)
+            || executeNumber<Int8>(src_data, src_offsets, res) || executeNumber<Int16>(src_data, src_offsets, res)
+            || executeNumber<Int32>(src_data, src_offsets, res) || executeNumber<Int64>(src_data, src_offsets, res)
+            || executeNumber<Float32>(src_data, src_offsets, res) || executeNumber<Float64>(src_data, src_offsets, res))
+        {
+            return res;
+        }
+        else
+            throw Exception(
+                ErrorCodes::ILLEGAL_COLUMN,
+                "Illegal column {} of first argument of function {}",
+                arguments[0].column->getName(),
+                getName());
+    }
+
+    template <typename T>
+    bool executeNumber(const IColumn & src_data, const ColumnArray::Offsets & src_offsets, ColumnPtr & res_ptr) const
+    {
+        const ColumnVector<T> * src_data_concrete = checkAndGetColumn<ColumnVector<T>>(&src_data);
+        if (!src_data_concrete)
+            return false;
+
+        const PaddedPODArray<T> & src_vec = src_data_concrete->getData();
+
+        auto outliers = ColumnFloat64::create();
+        auto & outlier_data = outliers->getData();
+
+        ColumnArray::ColumnOffsets::MutablePtr res_offsets = ColumnArray::ColumnOffsets::create();
+        auto & res_offsets_data = res_offsets->getData();
+
+        ColumnArray::Offset prev_src_offset = 0;
+        for (auto curr_src_offset : src_offsets)
+        {
+            chassert(prev_src_offset <= curr_src_offset);
+            size_t len = curr_src_offset - prev_src_offset;
+            if (len < 4)
+                throw Exception(ErrorCodes::BAD_ARGUMENTS, "At least four data points are needed for function {}", getName());
+
+            std::vector<Float64> src_sorted(src_vec.begin() + prev_src_offset, src_vec.begin() + curr_src_offset);
+            std::sort(src_sorted.begin(), src_sorted.end());
+
+            size_t q1_index = len / 4;
+            size_t q3_index = (len * 3) / 4;
+
+            Float64 q1 = (len % 2 != 0) ? src_sorted[q1_index] : (src_sorted[q1_index - 1] + src_sorted[q1_index]) / 2;
+            Float64 q3 = (len % 2 != 0) ? src_sorted[q3_index] : (src_sorted[q3_index - 1] + src_sorted[q3_index]) / 2;
+
+            Float64 iqr = q3 - q1;
```

Comment:
\+ `/// interquantile range`

Reply:
Afais, there is a chance that `iqr` becomes 0 (e.g. if all series elements are 0). Then, l. 107 will divide by 0 and ClickHouse goes boom.

We should fail gracefully, e.g. check + throw an exception or return 0 (not sure if the algorithm allows to do that).